File: src/main/java/io/github/oleksivio/telegram/bot/api/model/objects/inline/queryresult/InlineQueryResultCachedDocument.java
package io.github.oleksivio.telegram.bot.api.model.objects.inline.queryresult;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * @see <a href="https://core.telegram.org/bots/api#inlinequeryresultcacheddocument">InlineQueryResultCachedDocument</a>
 */
public class InlineQueryResultCachedDocument extends TitledInlineResult {
    private static final String TYPE = "document";
    /**
     * document_file_id String A valid file identifier for the file
     */
    @JsonProperty("document_file_id")
    private String document_file_id = null;
    /**
     * description String Optional. Short description of the result
     */
    @JsonProperty("description")
    private String description = null;

    InlineQueryResultCachedDocument() {
        super(TYPE);
    }

    public String getDocument_file_id() {
        return document_file_id;
    }

    public void setDocument_file_id(String document_file_id) {
        this.document_file_id = document_file_id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/objects/inline/queryresult/InlineQueryResultCachedDocument.kt
package io.github.oleksivio.telegram.bot.api.model.objects.inline.queryresult

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * @see [InlineQueryResultCachedDocument](https://core.telegram.org/bots/api.inlinequeryresultcacheddocument)
 */
class InlineQueryResultCachedDocument(
        /**
         * document_file_id String A valid file identifier for the file
         */
        @JsonProperty("document_file_id")
        var document_file_id: String? = null,
        /**
         * description String Optional. Short description of the result
         */
        @JsonProperty("description")
        var description: String? = null,
        /**
         * type String Type of the result
         */
        @JsonProperty("type")
        val type: String = "document"
) : TitledInlineResult()
